### PR TITLE
adding TALXIS Data Feed

### PR DESCRIPTION
**Pull request by honzakostejn**
---
#### Important: Our certification engineer, Srikanth, is currently on emergency leave. Please expect a delay with your PR. He will return January 31st, 2022. We apologize for any inconveniences.

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector works and I verified by deploying and testing all the operations.
image
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

### CODE REVIEW

**Comment by Amjed-Ayoub** [comment]
Dear partner,

We apologize for the delay you might be experiencing while we are working on a high volume of connector requests. Rest assured that we are doing everything we can to address this volume. We are taking measures like improving our processes and hiring additional team members to guarantee resolving your connector requests in a timely manner. 

Please let us know if you have any questions in the meantime.

File: certified-connectors/TALXISDataFeed/apiDefinition.swagger.json [hunk 1]
```diff
@@ -0,0 +1,1216 @@
...
+              "type": "number"
+            },
+            "lowerBound": {
+              "title": "Gross Revenue (Lower Bound)",
+              "format": "double",
+              "type": "number"
+            },
+            "year": {
+              "title": "Gross Revenue Year",
+              "description": "",
```
Comment (sriyen-msft) [inline certified-connectors/TALXISDataFeed/apiDefinition.swagger.json:102]: @honzakostejn Empty property values can be removed until unless you have a specific reason to keep them.

File: certified-connectors/TALXISDataFeed/apiProperties.json [hunk 2]
```diff
@@ -0,0 +1,112 @@
+{
+  "properties": {
+    "connectionParameters": {
+      "token": {
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "aad",
+          "clientId": "28d529aa-b85e-4469-9cf3-937bea582555",
```
Comment (sriyen-msft) [inline certified-connectors/TALXISDataFeed/apiProperties.json:8]: @honzakostejn Please remove the client id value as per Microsoft's security policy. You can replace it with dummy text.
`"clientId": "[[DUMMY]]"`

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Comment by sriyen-msft** [comment]
@honzakostejn I'm just checking in to see how things are going with updating your connector artifacts as per the given feedback. Please feel free to reach out with any questions you may have, Thanks.

File: certified-connectors/TALXISDataFeed/apiDefinition.swagger.json (hunk 1 above)
Comment (honzakostejn) [reply certified-connectors/TALXISDataFeed/apiDefinition.swagger.json:102]: I'll delete them.

File: certified-connectors/TALXISDataFeed/apiProperties.json (hunk 2 above)
Comment (honzakostejn) [reply certified-connectors/TALXISDataFeed/apiProperties.json:8]: We're using another application registration for the API. This one for connector has only rights to call the API itself. But I'll replace it.

**Comment by honzakostejn** [comment]
@sriyen-msft I'll be pushing the commit today / tomorrow. I'm adding one more connector so I want to make sure, that everything is okay.

**Comment by honzakostejn** [comment]
I'm not sure why the API Properties Validator is returning failed status. I don't see anything suspicious on lines 44... Could you help me out @sriyen-msft?

**Comment by v-eisljami** [comment]
In one PR only one connector can be submitted. Please add separate PRs for different connectors.

**Comment by Amjed-Ayoub** [comment]
Hi @honzakostejn, I hope you are doing well. It looks like you are updating multiple files for multiple connectors which is not allowed you can update only one connector per pull request.
I suggest you close this pull request and submit two separate pull requests for each connector.
Please let me know if you have any questions.
Thank you very much.

**Comment by Amjed-Ayoub** [comment]
Hi @honzakostejn, I hope you are doing well.
As you can see below you are updating two connectors in one pull request which is NOT allowed. May I ask you please to fix this. You can close this pull request and submit another two pull requests one for each connector.
image

Please let me know if you have any questions.

Thank you very much

**Comment by honzakostejn** [comment]
Hi @Amjed-Ayoub,

TALXIS Documents Connector was removed. However, the API properties validation is still failing... Could you provide me with the error message?

Thanks.

**Comment by Amjed-Ayoub** [comment]
Hi @honzakostejn, Please see below for errors.
##[error]@{level=Error; message=Invalid type. Expected Boolean but got String.; valuePath=line: 0 position: 0 path: 'properties.connectionParameters.token:clientId.uiDefinition.constraints.hidden'; value=true}
##[error]@{level=Error; message=Invalid type. Expected Boolean but got String.; valuePath=line: 0 position: 0 path: 'properties.connectionParameters.token:clientSecret.uiDefinition.constraints.hidden'; value=true}
##[error]@{level=Error; message=JSON does not match all schemas from 'allOf'. Invalid schema indexes: 0.; valuePath=line: 0 position: 0 path: 'properties.connectionParameters.token:TenantId.uiDefinition'; value=}
##[error]@{level=Error; message=Invalid type. Expected Boolean but got String.; valuePath=line: 0 position: 0 path: 'properties.connectionParameters.token:resourceUri.uiDefinition.constraints.hidden'; value=true}
##[error]@{level=Error; message=Invalid type. Expected Boolean but got String.; valuePath=line: 0 position: 0 path: 'properties.connectionParameters.token:grantType.uiDefinition.constraints.hidden'; value=true}
##[error]@{level=Critical; message=Invalid value for the description property within the UI Definition object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.description; value=The tenant ID of for the Azure Active Directory application}
##[error]@{level=Critical; message=Invalid value for the display name property within the UI Defintion object.; valuePath=$.properties.connectionParameters.token:TenantId.uiDefinition.displayName; value=Tenant}

I hope this helps.
Please let me know if you have any questions.

Thank you very much.

File: certified-connectors/TALXISDataFeed/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+* A Microsoft Power Apps or Power Automate plan
+* TALXIS license
+
+You will not be able to use this connector without TALXIS license. Contact NETWORG if you wish to get this license.
+
+### Set up Azure AD application
+We first need to register our connector as an application in Azure AD.  This will allow the connector to identify itself to Azure AD so that it can ask for permissions to access TALXIS Data Feed API on behalf of the end user.  You can read more about this [here](https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-scenarios) and follow the steps below:
+
+1. Approve 'TALXIS - Data Feed - Flow' Azure AD application
+This Azure AD application will be used to identify the connector against your Azure AD. [Approve the application registration in your tenant.](https://login.microsoftonline.com/common/adminconsent?client_id=28d529aa-b85e-4469-9cf3-937bea582555)
```
Comment (sriyen-msft) [inline certified-connectors/TALXISDataFeed/readme.md:15]: @honzakostejn It is not allowed to add any client credentials such as client id and secret in the connector artifacts, Can you please remove it from the readme.md file.

**Review by sriyen-msft — CHANGES_REQUESTED** [review]

**Comment by Amjed-Ayoub** [comment]
Hi @honzakostejn I hope you are doing well. I am following up with you to see how things are going with updating your artifacts.
Please let us know if you have any questions.
We look forward to getting your connector certified.
Thank you very much.

**Comment by Amjed-Ayoub** [comment]
Hello @honzakostejn,
This is to follow up on the status of your connector. I would like to offer help to you if you are having any issues with your connector.
Please do not hesitate to reach out if you have any questions as we are always happy to help.
Thank you very much for working with us.

**Comment by Amjed-Ayoub** [comment]
Hello @honzakostejn,
I hope you are doing well.
I am following up with you to see how things are going with preparing your connector.
Please let us know if you have any questions.
Thank you.

**Comment by Amjed-Ayoub** [comment]
Hello @honzakostejn,
I hope you are doing well.
I am following up in my messages above to see if you need any help with updating your connector artifacts.
Please do not hesitate to reach out to me if you have any questions as we are always here to help. 😊
I look forward to hearing from you.
Thank you very much.

**Comment by sriyen-msft** [comment]
@honzakostejn We are closing your PR since there is no response from your side. You can reopen the same PR whenever you are ready for certification of your connector, thanks.

**Comment by honzakostejn** [comment]
Hi,
Should I create a new one, or can you open this one again?

I'm looking into it right now.

**Comment by honzakostejn** [comment]
I have created a new PR #1875

**Pull request closed without merging**